HeatStroke/AudioPlayer/AudioPlayer.cpp

```diff
@@ -83,19 +65,66 @@ void HeatStroke::AudioPlayer::StopMusic()
 	}
 }
 
-void HeatStroke::AudioPlayer::PlaySoundEffectFromFile(const std::string& p_strFile)
+void HeatStroke::AudioPlayer::PreloadSoundEffects(const std::string& p_strPreloadFileName)
 {
-	sf::SoundBuffer* pSoundBuffer = new sf::SoundBuffer;
-	if (!(pSoundBuffer->loadFromFile(p_strFile)))
+	tinyxml2::XMLDocument mPreloadDoc;
+	assert(mPreloadDoc.LoadFile(p_strPreloadFileName.c_str()) == tinyxml2::XML_NO_ERROR);
+
+	tinyxml2::XMLElement* pRootElement = mPreloadDoc.RootElement();
+	assert(pRootElement != nullptr);
+
+	tinyxml2::XMLElement* pSoundEffectsElement = pRootElement->FirstChildElement("SoundEffects");
+	assert(pSoundEffectsElement != nullptr);
+
+	for (tinyxml2::XMLElement* pSoundEffectElement = pSoundEffectsElement->FirstChildElement("SoundEffect");
+		pSoundEffectElement != nullptr;
+		pSoundEffectElement = pSoundEffectElement->NextSiblingElement("SoundEffect"))
 	{
-		delete pSoundBuffer;
-		printf("Failed to play sound: %s\n", p_strFile.c_str());
+		std::string strSoundEffectFileName;
+		EasyXML::GetRequiredStringAttribute(pSoundEffectElement, "path", strSoundEffectFileName);
+
+		LoadedSoundEffects::const_iterator it = m_mLoadedSoundEffects.find(strSoundEffectFileName);
+		if (it == m_mLoadedSoundEffects.end())
+		{
+			sf::SoundBuffer* pSoundBuffer = new sf::SoundBuffer;
+			if (!(pSoundBuffer->loadFromFile(strSoundEffectFileName)))
+			{
+				delete pSoundBuffer;
+				printf("Failed to find sound: %s\n", strSoundEffectFileName.c_str());
+				return;
+			}
+
+			sf::Sound* pSound = new sf::Sound;
+			pSound->setBuffer(*pSoundBuffer);
+
+			m_mLoadedSoundEffects[strSoundEffectFileName] = std::pair<sf::SoundBuffer*, sf::Sound*>(pSoundBuffer, pSound);
+		}
+	}
+}
+
+void HeatStroke::AudioPlayer::PlaySoundEffect(const std::string& p_strFile)
+{
+	LoadedSoundEffects::const_iterator it = m_mLoadedSoundEffects.find(p_strFile);
+	
+	if (it == m_mLoadedSoundEffects.end())
+	{
+#ifdef _DEBUG
+		std::cout << "Sound file could not be played because it was not preloaded: " << p_strFile << "\n";
```

Comment:
I don't know about standard. I've been using std::cout since the beginning, and I was just not really bothering to correct people using printf.